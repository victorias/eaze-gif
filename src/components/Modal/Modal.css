.modal-content {
  display: flex;
  padding: 10px;
  outline: none;
  background-color: grey;
  max-width: 100vw;
}

.modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  background-color: rgba(255, 255, 255, 0.75);
}
