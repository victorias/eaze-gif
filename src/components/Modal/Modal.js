// @flow

import React from 'react';
import ReactModal from 'react-modal';
import styled from 'styled-components';

import './Modal.css';

type Props = {
  children?: React$Node,
  isOpen: boolean,
  onRequestClose: () => void,
};

const Modal = (props: Props) => (
  <ReactModal
    isOpen={props.isOpen}
    className="modal-content"
    overlayClassName="modal-overlay"
    contentLabel="Modal"
    onRequestClose={props.onRequestClose}
  >
    {props.children}
  </ReactModal>
);

export default Modal;
