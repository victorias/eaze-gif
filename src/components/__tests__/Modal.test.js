// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Modal from '../Modal';

describe('Modal', () => {
  it('renders when open', () => {
    const tree = renderer.create(
      <Modal isOpen onRequestClose={() => undefined} />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders when closed', () => {
    const tree = renderer.create(
      <Modal isOpen={false} onRequestClose={() => undefined} />
    );
    expect(tree).toMatchSnapshot();
  });
});
