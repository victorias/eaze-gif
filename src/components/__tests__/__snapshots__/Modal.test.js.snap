// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders when closed 1`] = `null`;

exports[`Modal renders when open 1`] = `null`;
